assets/app/scripts/directives/logViewer.js

```diff
@@ -224,6 +228,45 @@ angular.module('openshiftConsole')
               goChromeless: logLinks.chromelessLink,
               restartLogs: streamLogs
             });
+
+            APIDiscovery
+              .getLoggingURL()
+              .then(function(url) {
+                // TODO: update lodash for _.get, would rather do this:
+                // var projectName = _.get($scope, 'context', 'project', 'metadata', 'name');
+                // var containerName = _.get($scope, 'options', 'container');
+                var projectName = $scope.context &&
+                                  $scope.context.project &&
+                                  $scope.context.project.metadata &&
+                                  $scope.context.project.metadata.name;
+
+                var containerName = $scope.options &&
+                                    $scope.options.container;
+
+                if(!(projectName && containerName && $scope.name && url)) {
+                  return;
+                }
+
+                angular.extend($scope, {
+                  kibanaAuthUrl: $sce.trustAsResourceUrl(URI(url)
+                                                          .segment('auth').segment('token')
+                                                          .normalizePathname().toString()),
+                  access_token: AuthService.UserStore().getToken()
+                });
+
+                $scope.$watchGroup(['context.project.metadata.name', 'options.container', 'name'], function() {
```

Comment:
The watch won't be set if projectName / containerName / name is undefined on initialization.